app/Models/Campaign.php

```diff
@@ -252,53 +253,53 @@ public function getBasicInformationAttribute(): array
         ];
     }
 
-    public function changeStatus(int $status): void
+    public function changeStatus(int $status, ?ExchangeRate $exchangeRate = null): void
     {
         if ($status === $this->status) {
             return;
         }
 
         self::failIfInvalidStatus($status);
-        $this->failIfTransitionNotAllowed($status);
-        $this->updateBlockade($status);
+        $this->updateBlockadeOrFailIfNotAllowed($status, $exchangeRate);
 
         $this->status = $status;
     }
 
-    private function failIfTransitionNotAllowed(int $status): void
+    private function updateBlockadeOrFailIfNotAllowed(int $status, ?ExchangeRate $exchangeRate = null): void
     {
-        if ($status === self::STATUS_ACTIVE) {
-            $balance = UserLedgerEntry::getBalanceByUserId($this->user_id);
+        if ($status !== self::STATUS_ACTIVE) {
+            Log::info(sprintf('Hold the blockade'));
 
-            $requiredBalance = $this->getBudgetForCurrentDateTime();
-
-            if ($balance < $requiredBalance) {
-                throw new InvalidArgumentException('Campaign budgets exceed account balance');
-            }
+            return;
         }
-    }
 
-    private function updateBlockade(int $status): void
-    {
-        $budgetForCurrentDateTime = $this->getBudgetForCurrentDateTime();
-        if (0 === $budgetForCurrentDateTime) {
+        $budget = $this->getBudgetForCurrentDateTime();
+        if (0 >= $budget) {
             return;
         }
 
-        $amount = self::fetchRequiredBudgetForAllCampaignsInCurrentPeriod();
-        if ($status === self::STATUS_ACTIVE) {
-            $amount += $budgetForCurrentDateTime;
-        } elseif ($this->status === self::STATUS_ACTIVE) {
-            $amount -= $budgetForCurrentDateTime;
+        if (null === $exchangeRate) {
+            Log::error('ExchangeRate is not available');
+            throw new InvalidArgumentException('ExchangeRate is not available');
         }
 
+        $totalBudget = self::fetchRequiredBudgetForAllCampaignsInCurrentPeriod() + $budget;
```

Comment:
Does it mean that we didn't include the needed budget from the current hour before?

Reply:
Blockade update is done by releasing previous blockade.